torch/distributed/checkpoint/state_dict_saver.py

```diff
@@ -116,14 +120,19 @@ def save(
     torch._C._log_api_usage_once("torch.distributed.checkpoint.save")
 
     with _profile():
-        dumpable_state_dict = {}
-        for key, elem in state_dict.items():
-            dumpable_state_dict[key] = (
-                elem.state_dict() if isinstance(elem, Stateful) else elem
-            )
+        if not storage_writer:
```

Comment:
Nit: could be a good place for a helper function

Reply:
I think the `TODO` to decide whether to use `FSSpec` or `FileSystem` should be an util. But this is pretty standalone as StorageWriter and StorageReader are separate classes.